projectgenerator/gui/generate_project.py

```diff
@@ -117,6 +117,16 @@ def accepted(self):
 
         self.save_configuration(configuration)
 
+        cursor = QApplication.overrideCursor()
+        if cursor is None or cursor == 0:
+            QApplication.setOverrideCursor( Qt.WaitCursor )
+        elif cursor.shape() != Qt.WaitCursor:
+            QApplication.setOverrideCursor( Qt.WaitCursor )
```

Comment:
I think it's best to set the cursor without any check beforehand. What would be great but would need more refactoring is the following

```
try:
    QApplication.setOverrideCursor( ... )
    some code ...
finally:
    QApplication.restoreOverrideCursor()
```

That would be exception safe (I hate to see QGIS stuck with a wait cursor because some exception was raised that the dev didn't anticipate). But one needs to make sure that there is no return within the try-block.